test/jdk/java/net/httpclient/UserAuthWithAuthenticator.java

```diff
@@ -37,17 +49,122 @@
+import jdk.httpclient.test.lib.http2.Http2TestServer;
+import com.sun.net.httpserver.BasicAuthenticator;
 
 import jdk.test.lib.net.URIBuilder;
 
 import static java.nio.charset.StandardCharsets.US_ASCII;
 
 public class UserAuthWithAuthenticator {
 
+    static class AuthTestHandler implements HttpTestHandler {
+        volatile String authValue;
+        final String response = "Hello world";
+
+        @Override
+        public void handle(HttpTestExchange t) throws IOException {
+            try (InputStream is = t.getRequestBody();
+                 OutputStream os = t.getResponseBody()) {
+                byte[] bytes = is.readAllBytes();
+                authValue = t.getRequestHeaders().firstValue("Authorization").orElse("");
+                t.sendResponseHeaders(200, response.length());
+                os.write(response.getBytes(US_ASCII));
+                t.close();
+            }
+        }
+
+        String authValue() {return authValue;}
+    }
+
+    // if useHeader is true, we expect the Authenticator was not called
+    // and the user set header used. If false, Authenticator must
+    // be called and the user set header not used.
+
+    static void h2Test(final boolean useHeader) throws Exception {
+        SSLContext ctx;
+        HttpTestServer h2s = null;
+        HttpClient client = null;
+        ExecutorService ex=null;
+        try {
+            ctx = new SimpleSSLContext().get();
+            ex = Executors.newCachedThreadPool();
+            InetAddress addr = InetAddress.getLoopbackAddress();
+
+            h2s = HttpTestServer.of(new Http2TestServer(addr, "::1", true, 0, ex,
+                    10, null, ctx, false));
+            AuthTestHandler h = new AuthTestHandler();
+            var context = h2s.addHandler(h, "/test1");
+            context.setAuthenticator(new BasicAuthenticator("realm") {
+                public boolean checkCredentials(String username, String password) {
+                    if (useHeader) {
+                        return username.equals("user") && password.equals("pwd");
+                    } else {
+                        return username.equals("serverUser") && password.equals("serverPwd");
+                    }
+                }
+            });
+            h2s.start();
+
+            int port = h2s.getAddress().getPort();
+            ServerAuth sa = new ServerAuth();
+            var plainCreds = "user:pwd";
+            var encoded = java.util.Base64.getEncoder().encodeToString(plainCreds.getBytes(US_ASCII));
+
+            URI uri = URIBuilder.newBuilder()
+                 .scheme("https")
+                 .host(addr.getHostAddress())
+                 .port(port)
+                 .path("/test1/foo.txt")
+                 .build();
+
+            HttpClient.Builder builder = HttpClient.newBuilder()
+                    .sslContext(ctx)
+                    .executor(ex);
+
+            //if (!useHeader) {
```

Comment:
do you plan to keep this commented-out code?

Reply:
No, that needs to be removed.